src/opendr/perception/object_detection_2d/nanodet/nanodet_learner.py

```diff
@@ -333,11 +360,147 @@ def reset(self):
         """This method is not used in this implementation."""
         return NotImplementedError
 
-    def optimize(self):
-        """This method is not used in this implementation."""
-        return NotImplementedError
+    def _save_onnx(self, onnx_path, img=None, do_constant_folding=False, verbose=True):
+        if self.jit_model:
+            print("Warning: A jit model has already initialized, inference will run in onnx mode by default!!!")
+        if not self.predictor:
+            self.predictor = Predictor(self.cfg, self.model, device=self.device)
+
+        os.makedirs(onnx_path, exist_ok=True)
+        export_path = os.path.join(onnx_path, "nanodet_{}.onnx".format(self.cfg.check_point_name))
+
+        if self.dummy_input is None:
+            assert img is not None,\
+                "When use optimize or _save_jit is called for the first time, it must have and opendr Image input"
```

Comment:
As a more general comment, why is this the case? Why can't a dummy Tensor input be used as in other learners to optimize the model?
```suggestion
                "When optimize or _save_onnx is called for the first time, it must have and OpenDR image input."
```

Reply:
Nanodet is focus on inference speed. The input can be change its time in various shapes because it can have respect to width and height ratio.
For this reason it is better to give the exact input shape that it is going to have its time in inferenece and not let various input shapes so the optimizations can have better impact. This can be changed if we do not focus in faster performance. Or in the future I can have both implementations as different modes and specify if somone whould like to have better optimization with specific input sizes (better for most datasets) or easier use.

Reply:
I understand that it has to have a specific shape. However, what i meant is why it needs an *image* instead of a dummy tensor of the required shape, which is similar to other learners that optimize using ONNX.
An example can be seen [here](https://github.com/opendr-eu/opendr/blob/d7df097e77fb7a084a08c330dd1b4b36b4dfde87/src/opendr/perception/pose_estimation/lightweight_open_pose/lightweight_open_pose_learner.py#L862-L868).

Reply:
Maybe you can add width and height as arguments with some default values, so they can be changed when the method is called. My issue is using an Image to do it, which somewhat complicates things (image needs to be loaded etc)